File: spring/src/main/java/com/example/shoppingCart/models/CartDetail.java
package com.example.shoppingCart.models;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Data
public class CartDetail {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private int quantity;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime dateModified;

    @ManyToOne
    private Cart cart;

    @ManyToOne
    private Item item;
}

File: Submission/angular/shopping-cart/src/app/models/CartDetail.ts
import { ICart } from './Cart';
import { Item, IItem } from './Item';

export interface ICartDetail {
    id: number;
    quantity: number;
    dateModified: number;
    cart: ICart;
    item: IItem;
}

export class CartDetail implements ICartDetail {
    id: number = 0;
    quantity: number;
    dateModified: number;
    cart: ICart;
    item: IItem;
}